packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -459,34 +458,89 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
         });
     }
 
-    private async fetchSnapshot(snapshotOptions: string, tokenFetchOptions: TokenFetchOptions) {
-        const storageToken = await this.getStorageToken(tokenFetchOptions, "TreesLatest");
+    private async fetchSnapshot(snapshotOptions: ISnapshotOptions, tokenFetchOptions: TokenFetchOptions) {
+        const usePost = this.hostPolicy.usePostForTreesLatest;
+        // If usePost is false, then make get call for TreesLatest.
+        // If usePost is true, and token refresh is false, then it means we are trying it first time, so
+        // don't catch the error as getWithRetryForTokenRefresh will handle and try with refresh.
+        // If usePost is true and we are trying with a refreshed token, then use the fallback code in catch.
+        if (usePost && tokenFetchOptions.refresh) {
+            try {
+                return this.fetchSnapshotCore(snapshotOptions, tokenFetchOptions, true);
+            } catch (error) {
+                this.logger.sendErrorEvent({ eventName: "TreeLatest_FallBackToGetRequest" });
+                return this.fetchSnapshotCore(snapshotOptions, tokenFetchOptions, false);
+            }
+        } else if (usePost) {
+            return this.fetchSnapshotCore(snapshotOptions, tokenFetchOptions, true);
+        } else {
+            return this.fetchSnapshotCore(snapshotOptions, tokenFetchOptions, false);
+        }
+    }
 
-        // TODO: This snapshot will return deltas, which we currently aren't using. We need to enable this flag to go down the "optimized"
-        // snapshot code path. We should leverage the fact that these deltas are returned to speed up the deltas fetch.
-        const { headers, url } = getUrlAndHeadersWithAuth(`${this.snapshotUrl}/trees/latest${snapshotOptions}`, storageToken);
+    private async fetchSnapshotCore(
+        snapshotOptions: ISnapshotOptions,
+        tokenFetchOptions: TokenFetchOptions,
+        usePost: boolean,
+    ) {
+        const storageToken = await this.getStorageToken(tokenFetchOptions, "TreesLatest");
+        let url: string;
+        let headers: {[index: string]: string};
+        let postBody: string;
+        if (usePost) {
+            url = `${this.snapshotUrl}/trees/latest?ump=1`;
+            const formBoundary = uuid();
+            const crlf = "\r\n";
+            postBody = `--${formBoundary}${crlf}`;
+            postBody += `Authorization: Bearer ${storageToken}${crlf}`;
+            postBody += `X-HTTP-Method-Override: GET${crlf}`;
+            Object.entries(snapshotOptions).forEach(([key, value]) => {
+                postBody += `${key}: ${value}${crlf}`;
+            });
+            postBody += `post: 1${crlf}`;
```

Comment:
Payload looks weird.
Why do we have guids around it?
Why do we need "post: 1"? I'm sure the fact that POST is used is available on the other side (though it might be a limitation of SPO where this info is lost in translation).

I'd inline crcl - code would be more readable.

Reply:
A multi part form request looks like this where we use guids to make the form boundary. As confirmed with Ram, we can use _POST as they use _ to define internal vs external. They want to use this param for telemetry purpose so as to check whether they are getting any Options call around it.
Inlined crlf